Sources/PayPal/PayPalUIInterfaces/PayPalUIFlow.swift

```diff
@@ -0,0 +1,52 @@
+import Foundation
+import UIKit
+@_implementationOnly import PayPalCheckout
+
+#if canImport(PaymentsCore)
+import PaymentsCore
+#endif
+
+protocol PayPalUIFlow {
+    typealias CreateOrderCallback = (PayPalCreateOrder) -> Void
+    typealias ApprovalCallback = (PayPalCheckoutApprovalData) -> Void
+    typealias CancelCallback = () -> Void
+    typealias ErrorCallback = (PayPalCheckoutErrorInfo) -> Void
+
+    static func set(config: CoreConfig, returnURL: String)
+
+    static func start(
+        presentingViewController: UIViewController?,
+        createOrder: CreateOrderCallback?,
+        onApprove: ApprovalCallback?,
+        onCancel: CancelCallback?,
+        onError: ErrorCallback?
+    )
+}
+
+extension Checkout: PayPalUIFlow {
+    static func set(config: CoreConfig, returnURL: String) {
+        let nxoConfig = CheckoutConfig(
+            clientID: config.clientID,
+            returnUrl: returnURL,
+            environment: config.environment.toNativeCheckoutSDKEnvironment()
```

Comment:
Did you mess around with removing `set` from the `PayPalUI` protocol and this extension? Ideally `config.environment.toNativeCheckoutSDKEnvironment()` (business logic) doesn't live in this extension, but rather in `PayPalClient`. 

I wouldn't be surprised if this is something you explored, but ran into issues abstracting the `CheckoutConfig` initializer. I _think_ we could achieve this by also abstracting `CheckoutConfig` in a similar way to the above - but would need to mess with it to be sure.

Reply:
Yeah i wanted to avoid making the protocol generic - so i had to conform the `CheckoutConfig` type to a protocol that only involved core types. I feel like the `toNativeCheckoutSDKEnvironment` conversion isn't so much business logic as it is interface conformance similar to the other conformances in the `PayPalUIInterfaces` directory. what do you think? (sorry for my delay getting back to this)